feature-cancel-insurance/src/main/kotlin/com/hedvig/android/cancelinsurance/CancelInsuranceViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package com.hedvig.android.cancelinsurance
+
+import androidx.compose.material3.DatePickerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import arrow.core.Either
+import com.hedvig.android.cancelinsurance.data.CancelInsuranceUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(ExperimentalMaterial3Api::class)
+internal class CancelInsuranceViewModel(
+  private val insuranceId: InsuranceId,
+  private val cancelInsuranceUseCase: CancelInsuranceUseCase,
+  clock: Clock = Clock.System,
+) : ViewModel() {
+  private val dateSubmissionError = MutableStateFlow(false)
```

Comment:
Why not keep these in the viewstate?

Reply:
These are part of the UIstate through the `combine` done a few lines down from here.
(btw, uistate <-> viewstate, I use those interchangeably. I do think I like UIstate more lately though, as ViewState makes me think of the XML view, and UI-state for UI in compose, so I tend to use uiState instead, I hope that's not confusing)

Basically the reason they are separate is so that I can individually set the to true/false without having to take the existing UIState, copy it with this value changed, and set it back again.
Now to switch the error you just do `dateSubmissionError.update { true }` and `combine` triggers automatically and produces the new UiState.
With all of it being in a local _viewState it'd be 
```
_viewState.update { viewState ->
  viewState.copy(
    dateSumissionError = true
  )
}
```

This achieves the same result but in a pretty simple way imo. Basically like having all the puzzle pieces individually, and then putting them all together nicely in a UIState through the combine.

What do you think about it in general, does it make sense?

Reply:
I like naming it UiState as well! 

Regarding separate flows for state - for me it makes it a bit harder to follow since I have to know that eg. `dateSubmissionError` will get combined and result in `CancelInsuranceUiState`. And that `isPerformingCancelInsuranceNetworkRequest` translates to `isLoading` in the UiState. 

Its a bit clearer when you only have one state class that gets manipulated directly.